File: calculate-PI-by-Monte-Carlo-method.jl
number_of_point = parse(Int, ARGS[1])

not_inside_circle = 0

for i in 1:number_of_point
    x = 2*rand() - 1
    y = 2*rand() - 1
    if (x^2 + y^2) > 1
        not_inside_circle = not_inside_circle + 1
    end
end

inside_circle = number_of_point - not_inside_circle

println("number_of_point : $number_of_point")
println("inside_circle : $inside_circle")
println("not_inside_circle : $not_inside_circle")

println("==========================================================================================")

println((inside_circle * 4) / number_of_point)

File: calculate-PI-by-Monte-Carlo-method.py
import sys
sys.set_int_max_str_digits(2147483647)

number_of_point = int(sys.argv[1])

from decimal import Decimal, getcontext
from fractions import Fraction

getcontext().prec = 1000

import random

not_inside_circle = 0
for i in range(0, number_of_point):
  if ( (((random.uniform(0, 2) - 1)**2) + ((random.uniform(0, 2) - 1)**2)) > 1 ):
    not_inside_circle = not_inside_circle + 1

inside_circle = number_of_point - not_inside_circle

print(f"number_of_point : {number_of_point}")
print(f"inside_circle : {inside_circle}")
print(f"not_inside_circle : {not_inside_circle}")

print("==========================================================================================")

fraction = Fraction(inside_circle * 4, number_of_point)
print(Decimal(fraction.numerator) / Decimal(fraction.denominator))
